sklearn/decomposition/_nmf.py

```diff
@@ -1330,6 +1338,11 @@ def fit(self, X, y=None, **params):
         -------
         self
         """
+        if self.init == 'warn':
+            warnings.warn(("The default value of init for solver 'mu' will "
```

Comment:
This warning message is not complete. The default behavior (init=None) is:
- "nndsvd" **if n_components <= min(n_samples, n_features),**
- "random" otherwise.

I agree this condition is True in virtually all practical cases, but we should precise it in the warning.